src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,34 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
+    static const struct sol_str_table table[] = {
```

Comment:
You should check if edge is not NULL. Otherwise `sol_str_slice_from_str(edge)` will crash!